### PR TITLE
ENT system improvement

### LINKED ISSUE #76: Add ENT Val d'Oise
Hello,
Is it possible to add the connection for the ENT Val d'Oise : https://cas.moncollege.valdoise.fr/
Thanks

### LINKED ISSUE #110: ENT  monbureaunumerique
It doesn't work , but when i take alook at the code i think i find the error here :
 # ENT / PRONOTE required URLs
    ent_login_page = "https://cas.monbureaunumerique.fr/login?selection=EDU&service=http%3A%2F%2Fpronote.lycee-fabert.com%2Fpronote%2F&submit=Valider"
    ent_load = "https://educonnect.education.gouv.fr/idp/profile/SAML2/POST/SSO"
    ent_login = "https://educonnect.education.gouv.fr/idp/profile/SAML2/POST/SSO?execution=e1s1"
    pronote_verif = "https://cas.monbureaunumerique.fr/saml/SAMLAssertionConsumer"

Ent_login_page is not for my school , do i need to change it from my login url ?

https://cas.monbureaunumerique.fr/login?service=https%3A%2F%2Fclg-jamyn.monbureaunumerique.fr%2Fsg.do%3FPROC%3DPAGE_ACCUEIL

### LINKED ISSUE #119: Attachements (links) of homework?
Hello, I'm wondering if there's a way to check links in homework

### PR DESCRIPTION
**Pull request by Bapt5**
* add ENT Val d'Oise (fix #76)
* Make ENT system more readable
* add some ENT (fix #110)
* Document files and links (fix #119)

### CODE REVIEW

**Comment by Bapt5** [comment]
Why are you forcing formatting with black this is nitpicky, annoys developers and does not improve efficiency?

**Comment by Bapt5** [comment]
> I agree black is sometimes annoying but it does all of the work for you, so it's just a matter of getting used to it. It makes sure all the codebase looks the same with multiple contributors. Its really as simple as doing: `black pronotepy` in the repo root before you commit.

I made the ENT system more readable but running black complicates everything

**Comment by bain3** [comment]
I'm not sure what's wrong with the black formatted code. Looks good to me, no?

**Comment by Bapt5** [comment]
I think it is a bad idea to make private function public. For me mypy work well. Why don't we use pep8 for formatting instead of black?

File: pronotepy/dataClasses.py [hunk 1]
```diff
@@ -803,6 +803,8 @@ class Homework(Object):
         if the homework is marked done
     date : datetime.date
         deadline
+    files : List[File]
+        files and links attached to the homework
```
Comment (bain3) [inline pronotepy/dataClasses.py:807]: This is already documented since it's a property.

**Review by bain3 — COMMENTED** [review]
Just that one thing LGTM

**Comment by bain3** [comment]
Sorry but we won't switch to anything else, at least not autopep8. With black you just need to get used to it. It takes off any responsibility with formatting, you just run black before you push and that's it. The function arguments are a little weird but it makes the line lengths bearable.

File: pronotepy/dataClasses.py (hunk 1 above)
Comment (Bapt5) [reply pronotepy/dataClasses.py:807]: yes you're right I will change

**Comment by Bapt5** [comment]
> Sorry but we won't switch to anything else, at least not autopep8. With black you just need to get used to it. It takes off any responsibility with formatting, you just run black before you push and that's it. The function arguments are a little weird but it makes the line lengths bearable.

Okay I understand. Do you know if there is a way to auto black format on commit ?

**Comment by bain3** [comment]
Look at pre-commit hooks

**Pull request merged**